src/components/page/WithGATracker/index.tsx

```diff
@@ -30,32 +43,47 @@ export const getGAusername = (): string => username;
  * @param {string} page the url string of the page view being tracked
  * @param {FlexObject} options tracking options for the page view
  */
-export const trackPage = (page: string, options: FlexObject = {}): void => {
-  GoogleAnalytics.set({
-    page,
-    ...options,
-  });
-  GoogleAnalytics.pageview(page);
+export const trackPage = (page: string, options?: FlexObject): void => {
+  const gaCode = getGaCode(options);
+  if (gaCode && gaCode.length) {
+    GoogleAnalytics.set({
+      page,
+      ...options,
+    });
+    GoogleAnalytics.pageview(page);
+  }
 };
 
-if (GA_CODE.length) {
-  GoogleAnalytics.initialize(GA_CODE, {
-    testMode: GA_ENV === 'test',
-  });
-  GoogleAnalytics.set({
-    env: GA_ENV,
-    username,
-  });
-}
+/**
+ * helper function to initialize GoogleAnalytics
+ * @param {FlexObject} options tracking options for the page view
+ */
+export const initGoogleAnalytics = (options?: FlexObject) => {
+  const gaCode = getGaCode(options);
+  if (gaCode && gaCode.length) {
+    GoogleAnalytics.initialize(GA_CODE, {
+      testMode: GA_ENV === 'test',
+    });
+    GoogleAnalytics.set({
+      env: GA_ENV,
+      username,
+    });
+  }
+};
```

Comment:
- I still maintain that `options` should not be a FlexObject.
- I think that you should not get `GA_ENV` from the global scope in this module.  Is there a compelling reason why we don't simple get it from `options` or another param supplied to `initGoogleAnalytics`?  A nice side effect of doing so would be to make `initGoogleAnalytics` more flexible AND testable.

Reply:
We can pass `GA_ENV` in through the `options` param.